docs/design/w3c/w3c-representation.md

```diff
@@ -0,0 +1,886 @@
+    cred: ObjectHandle,
+    id: FfiStr,
+    cred_p: *mut ObjectHandle,
+) -> ErrorCode {}
+```
+
+#### Optional: Presentation Conversion methods
+
+Presentation conversion methods are only required if we decide not to implement duplication flow methods even for
+presentation exchange.   
+In this case, library will provide APIs to create/verify legacy formatted presentation and APIs to convert it into/from
+W3C form to support different Verifiers.
+
+```rust
+/// Convert legacy styled AnonCreds presentation into W3C AnonCreds presentation form
+///     The conversion process described at the specification: ---
+///
+/// # Params
+/// cred -      object handle pointing to legacy styled AnonCreds presentation to convert
+/// cred_p -    reference that will contain converted presentation (in W3C form) instance pointer
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_presentation_to_w3c(
+    cred: ObjectHandle,
+    cred_p: *mut ObjectHandle,
+) -> ErrorCode {}
+
+/// Convert W3C styled AnonCreds presentation into legacy styled AnonCreds presentation
+///     The conversion process described at the specification: ---
+///
+/// # Params
+/// cred -      object handle pointing to W3C styled AnonCreds presentation to convert
+/// cred_p -    reference that will contain converted presentation (in legacy form) instance pointer
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_presentation_from_w3c(
+    cred: ObjectHandle,
+    cred_p: *mut ObjectHandle,
+) -> ErrorCode {}
+```
+
+#### Optional: Methods duplication
+
+The idea for this approach to duplicate all issuance/presentation related methods for w3c standard.
+So the credentials and presentations themselves are generated using new flow methods.
+
+
+> Note, that we still need to implement credential conversion methods to support the case of using existing/issued
+> credentials for doing the w3c presentation.
+> Firstly, old credentials should be converted into a W3C form, and the next new presentation creation method can be
+> used.
+
+In fact, there are 6 main flow methods in total.
+4 of them we have to duplicate: `create_credential`, `process_credential`, `create_presentation`, `verify_presentation`.
+Whether we want to duplicate `create_offer`, `create_request` methods if we do not change their format?
+
+The reasons for adding duplication methods:
+
+- avoid breaking changes in the existing API
+    - for example if we want Credential Offer pointing to the form of a credential to be issued
+- clear separation between flows
+    - if a flow targeting issuing of W3C Credential the specific set of function to be used
+- avoid the situation when function result value may be in different forms
+    - example:
+        - issuer creates offer in legacy form but with resulting credential format indication (legacy or w3c )
+        - as the flow execution result, create credential function returns credential either in w3c or legacy form
```

Comment:
is it possible they will want both?

Reply:
Yes. Both versions can be used depending on the higher level issuance protocol version